shunting-yard-replicator/src/test/java/com/hotels/shunting/yard/replicator/exec/messaging/MessageReaderAdapterTest.java

```diff
@@ -0,0 +1,218 @@
+    when(dropPartitionEvent.getEventType()).thenReturn(EventType.ON_DROP_PARTITION);
+    when(dropPartitionEvent.getDeleteData()).thenReturn(true);
+
+    MetaStoreEvent actual = messageReaderAdapter.next();
+
+    MetaStoreEvent expected = MetaStoreEvent
+        .builder(EventType.ON_DROP_PARTITION, TEST_DB, TEST_TABLE)
+        .partitionColumns(PARTITION_COLUMNS)
+        .partitionValues(PARTITION_VALUES)
+        .deleteData(true)
+        .environmentContext(EMPTY_MAP)
+        .build();
+
+    assertMetaStoreEvent(actual, expected);
+
+  }
+
+  @Test
+  public void dropTableEvent() {
+    when(messageReader.next()).thenReturn(dropTableEvent);
+    configureMockedEvent(dropTableEvent);
+    when(dropTableEvent.getEventType()).thenReturn(EventType.ON_DROP_TABLE);
+    when(dropTableEvent.getDeleteData()).thenReturn(true);
+
+    MetaStoreEvent actual = messageReaderAdapter.next();
+
+    MetaStoreEvent expected = MetaStoreEvent
+        .builder(EventType.ON_DROP_TABLE, TEST_DB, TEST_TABLE)
+        .deleteData(true)
+        .environmentContext(EMPTY_MAP)
+        .build();
+
+    assertMetaStoreEvent(actual, expected);
+  }
+
+  @Test
+  public void insertEvent() {
+    Map<String, String> partitionKeyValues = IntStream
+        .range(0, partitionKeys.size())
+        .collect(LinkedHashMap::new,
+            (m, i) -> m.put(partitionKeys.get(i).getName(), partitionValues.get(0).getValues().get(i)), Map::putAll);
+
+    when(messageReader.next()).thenReturn(insertEvent);
+    configureMockedEvent(insertEvent);
+    when(insertEvent.getKeyValues()).thenReturn(partitionKeyValues);
+    when(insertEvent.getEventType()).thenReturn(EventType.ON_INSERT);
+
+    MetaStoreEvent expected = MetaStoreEvent
+        .builder(EventType.ON_INSERT, TEST_DB, TEST_TABLE)
+        .partitionColumns(PARTITION_COLUMNS)
+        .partitionValues(PARTITION_VALUES)
+        .environmentContext(EMPTY_MAP)
+        .parameters(EMPTY_MAP)
+        .build();
+
+    MetaStoreEvent actual = messageReaderAdapter.next();
+
+    assertMetaStoreEvent(actual, expected);
+
+  }
+
+  @Test
+  public void ignoresCreateTableEvent() {
+    when(messageReader.next()).thenReturn(createTableEvent);
+    configureMockedEvent(createTableEvent);
+    when(createTableEvent.getEventType()).thenReturn(EventType.ON_CREATE_TABLE);
+
+    MetaStoreEvent expected = MetaStoreEvent
+        .builder(EventType.ON_CREATE_TABLE, TEST_DB, TEST_TABLE)
+        .environmentContext(EMPTY_MAP)
+        .parameters(EMPTY_MAP)
+        .build();
+
+    MetaStoreEvent actual = messageReaderAdapter.next();
+
+    assertMetaStoreEvent(actual, expected);
+
+  }
+
+  private void configureMockedEvent(SerializableListenerEvent serializableListenerEvent) {
+    when(serializableListenerEvent.getDatabaseName()).thenReturn(TEST_DB);
+    when(serializableListenerEvent.getTableName()).thenReturn(TEST_TABLE);
+  }
+
+  private void assertMetaStoreEvent(MetaStoreEvent actual, MetaStoreEvent expected) {
```

Comment:
Swap these paramaters around to match Junit assertions which are in the order "expected" then "actual".